src/test/java/relay/data_access/InMemoryInstructorDataAccessObject.java

```diff
@@ -0,0 +1,46 @@
+package relay.data_access;
+
+import relay.entity.Instructor;
+import relay.exceptions.ResourceAlreadyExistsException;
+import relay.exceptions.ResourceNotFoundException;
+import relay.use_case.login.LoginInstructorDataAccessInterface;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * In-memory Instructor Data Access Object for testing purposes.
+ * This class provides methods for reading and saving instructor data in memory.
+ */
+public class InMemoryInstructorDataAccessObject implements LoginInstructorDataAccessInterface {
+
+    // Storage for instructor data in memory
+    private final Map<String, Instructor> instructors = new HashMap<>();
+
+    /**
+     * Retrieves an instructor by their email address.
+     *
+     * @param emailAddress The email address of the instructor to retrieve.
+     * @return The instructor associated with the given email address.
+     * @throws ResourceNotFoundException if no instructor is found with the provided email address.
+     */
+    @Override
+    public Instructor readByEmailAddress(String emailAddress) throws ResourceNotFoundException {
+        for (Instructor instructor : instructors.values()) {
+            if (instructor.getEmailAddress().equals(emailAddress)) {
+                return instructor;
+            }
+        }
+        throw new ResourceNotFoundException("No such instructor exists");
+    }
+
+    /**
+     * Saves an instructor to the in-memory data storage.
+     *
+     * @param instructor The instructor to be saved.
+     * @throws ResourceAlreadyExistsException if an instructor with the same ID already exists.
+     */
+    public void save(Instructor instructor) throws ResourceAlreadyExistsException {
+        instructors.put(instructor.getInstructorID(), instructor);
+    }
+}
```

Comment:
Not sure what's happening here. The readByEmailAddress should just be inside the FirebaseInstructorDataAccessObject. We have no need to save it into memory and then retrieve from that. Not only would that cause lots of memory usage but Firebase already manages it so it's much easier